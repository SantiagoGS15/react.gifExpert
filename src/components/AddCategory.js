import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
  
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue?.trim().length > 2){
            setCategories(categories => [inputValue, ...categories]);
            setInputValue('');
        }
    }

    return (
    <form onSubmit={handleSubmit}>
        <h1>Add Category</h1>
        <input 
            value={inputValue}
            onChange={handleInputChange}

            type= "text"
        />
    </form>);
};


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};