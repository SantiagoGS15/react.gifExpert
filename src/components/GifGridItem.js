import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {
  return (<div className="card">
      <img src={url} alt={title}></img>
      <p className="animate__animated animate__bounce">{title}</p>
  </div>);
};


GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};