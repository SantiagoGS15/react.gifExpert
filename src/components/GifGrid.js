import React from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchGifs(category);
  
    return (
      <>
        <h3 className="animate__fadeIn">{category}</h3>
        {loading && <p>Loading...</p>}
        <div className="card-grid">
          
          <ol>
            {
              images.map((item) => {
                return <GifGridItem 
                      {...item}
                      key={item.id}>
                    </GifGridItem>
              })
            }
          </ol>
        </div>
      </>
  );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};
