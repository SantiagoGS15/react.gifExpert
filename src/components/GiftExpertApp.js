import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';
import PropTypes from 'prop-types';

export const GiftExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);
    
    return (<>
        <h2>GiftExpertApp</h2>
        <AddCategory setCategories={setCategories}></AddCategory>
        <hr></hr>
        
        <ol>
            {
               categories.map((category) => {
                   return <GifGrid category={category} key={category}/>
               })
            }
        </ol>
    </>);
};

GiftExpertApp.propTypes = {
    defaultCategories: PropTypes.array
}