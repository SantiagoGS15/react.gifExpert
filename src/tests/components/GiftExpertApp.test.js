import { shallow } from 'enzyme';
import { GiftExpertApp } from '../../components/GiftExpertApp';
import React from 'react';

describe('Pruebas en GiftExpertApp', () => {

    test('Se debe mostrar correctamente', () => {
        const wrapper = shallow(<GiftExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Se debe mostrar una lista de caterogÃ­as', () => {
        const categories = ['One Punch', 'Dragon Ball']
        const wrapper = shallow(<GiftExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
});