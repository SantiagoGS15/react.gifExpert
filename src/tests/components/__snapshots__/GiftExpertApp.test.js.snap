// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftExpertApp Se debe mostrar correctamente 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en GiftExpertApp Se debe mostrar una lista de caterog√≠as 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
